### PR TITLE
Add SASL_LIBRARY link to all sub projects.

### PR DESCRIPTION
**Pull request by niyue**
Currently, if SASL_LIBRARY is needed, several sub projects cannot be built because SASL_LIBRARY is not added to the these targets' linked libraries list, and linking will fail.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -197,6 +197,18 @@ if (BUILD_OPTION_GEN_COVERAGE)
...
+#---------------------------
+if(DEFINED ENV{SASL_LIBRARYDIR})
+    set(SASL_LIBRARYDIR $ENV{SASL_LIBRARYDIR})
+    link_directories(${SASL_LIBRARYDIR})
+    message(STATUS "sasl2 library directory: ${SASL_LIBRARYDIR}")
+endif()
+if(DEFINED ENV{SASL_LIBRARY})
+    set(SASL_LIBRARY $ENV{SASL_LIBRARY})
+    message(STATUS "sasl2 library: ${SASL_LIBRARY}")
+endif()
```
Comment (niyue) [inline CMakeLists.txt:211]: This is moved from `tools/CMakeLists.txt`

**Comment by kenneth-jia** [comment]
The `modern-cpp-kafka` is just a header-only C++ wrapper, and the user applications/projects should decide whether to link sasl lib in, or not.
Besides `tools`, there're also binaries built for `tests`, while we don't need it for CI.
So, do you mind providing more details about your failure? I.e, are you include the complete `modern-cpp-kafka` directory in your project?

**Comment by niyue** [comment]
Hi @kenneth-jia,

I am not trying to include `modern-cpp-kafka` in my project for this, instead, I would like to add some new APIs for `modern-cpp-kafka`, and would like to build the project, including unit tests, tools and examples. But when I try to do some development for this project, my local env happens to have a librdkafka with sasl. Since currently, `modern-cpp-kafka`'s unit test doesn't enable sasl, in this case, if developers would like to build this project, they have to have two envs, which is not convenient. 

In this PR, I added `${SASL_LIBRARY}` to unit tests and examples modules, but since this `${SASL_LIBRARY}` is optional (it will be empty if not specified), so for users not using this feature, they won't be affected, and I hope this makes it easier to develop for this project if SASL is used by the developer.

File: examples/CMakeLists.txt [hunk 2]
```diff
@@ -2,23 +2,23 @@ project("kafka-examples")
 
 # Target: kafka_sync_producer
 add_executable("kafka_sync_producer" "kafka_sync_producer.cc")
-target_link_libraries("kafka_sync_producer" modern-cpp-kafka-api)
+target_link_libraries("kafka_sync_producer" modern-cpp-kafka-api ${SASL_LIBRARY})
```
Comment (niyue) [inline examples/CMakeLists.txt:5]: If `${SASL_LIBRARY}` is not specified, it will be empty and it won't affect existing usage.

**Comment by kenneth-jia** [comment]
I see your points, and it also makes sense to test Kafka with sasl enabled.
Let's merge it into a feature branch (i.e. `Feature-EnableCIForTestWithSASLConnection`) as the first step, I'll refine the CI to enable some test with sasl enabled, and merge the feature branch to `main` when all is ready.

**Pull request merged**